src/ServerDaemon.chpl

```diff
@@ -716,28 +736,366 @@ module ServerDaemon {
                         processMetrics(user, cmd, msgArgs, elapsedTime, memUsed);
                     }
                 }
-            } catch (e: ErrorWithMsg) {
+            } catch (ewm: ErrorWithMsg) {
                 // Generate a ReplyMsg of type ERROR and serialize to a JSON-formatted string
-                sendRepMsg(serialize(msg=e.msg,msgType=MsgType.ERROR, msgFormat=MsgFormat.STRING, 
+                sendRepMsg(serialize(msg=ewm.msg,msgType=MsgType.ERROR, msgFormat=MsgFormat.STRING, 
                                                         user=user));
+                if metricsEnabled() {
+                    processErrorMessageMetrics(user, cmd, ewm);
+                }                
+
                 if trace {
                     sdLogger.error(getModuleName(),getRoutineName(),getLineNumber(),
-                        "<<< %s resulted in error %s in  %.17r sec".doFormat(cmd, e.msg, timeSinceEpoch().totalSeconds() - s0));
+                        "<<< %s resulted in error %s in  %.17r sec".doFormat(cmd, 
+                                                                             ewm.msg, 
+                                                                             timeSinceEpoch().totalSeconds() - s0));
+                }
+            } catch (err: OverMemoryLimitError) {
+                var errorMsg = err.message();
```

Comment:
There seems like a lot of code duplication here....  Is it possible to make a method that can be called for each Error type?

Reply:
@ajpotts good point and sadly, no, at least to the best of my knowledge. Chapel casts all errors to Error, so I can't get the error types otherwise. If there's a better way, definitely interested in it, 'cause I agree, the code dupe is non-good.

Reply:
OK, thanks.